### PR TITLE
Support: Debug Adapter Protocol

### PR DESCRIPTION
**Pull request by david-driscoll**
@TylerLeonhardt  Just getting started on the DAP stuff... went pretty fast for the receiver end.

see: #145

### CODE REVIEW

File: src/JsonRpc/Serialization/SerializerBase.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        public T DeserializeObject<T>(string json)
+        {
+            return JsonConvert.DeserializeObject<T>(json, Settings);
+        }
+        public long GetNextId()
+        {
+            lock (_lock)
+            {
```
Comment (mholo65) [inline src/JsonRpc/Serialization/SerializerBase.cs]: Use `Interlocked.Increment` instead.

File: src/Protocol/Serialization/Serializer.cs [hunk 2]
```diff
@@ -159,5 +169,12 @@ public void SetClientCapabilities(ClientVersion clientVersion, ClientCapabilitie
                 workspaceSymbolKinds
             );
         }
+        public long GetNextId()
+        {
+            lock (_lock)
```
Comment (mholo65) [inline src/Protocol/Serialization/Serializer.cs]: Use `Interlocked.Increment` instead.

**Comment by TylerLeonhardt** [comment]
@david-driscoll it looks like this PR has everything I need in order to use `JsonRpcServer` as the `DebugServer` of sorts. Let me know I can help so I can start working off of this.

**Comment by TylerLeonhardt** [comment]
Alright, this feels done. Just need to get tests passing

File: src/Dap.Protocol/Requests/SetBreakpointsArguments.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        /// <summary>
+        /// The code locations of the breakpoints.
+        /// </summary>
+        [Optional] public Container<SourceBreakpoint> Breakpoints { get; set; }
+
+        /// <summary>
+        /// Deprecated: The code locations of the breakpoints.
+        /// </summary>
+        [Obsolete("Deprecated")]
+        [Optional] public Container<long> Lines { get; set; }
```
Comment (andschwa) [inline src/Dap.Protocol/Requests/SetBreakpointsArguments.cs:24]: @david-driscoll @TylerLeonhardt This seems to have been written deprecated...what should be used in place of it?

**Pull request merged**